Race condition during gossip startup

I experienced a Pilosa node crashing on startup of a 3 node cluster with the following stack trace:
```
2017/08/21 21:35:08 [DEBUG] memberlist: Initiating push/pull sync with: 10.0.1.167:12000
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x990fcb]

goroutine 24 [running]:
github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*TransmitLimitedQueue).GetBroadcasts(0x0, 0x3, 0x447, 0x0, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/queue.go:72 +0x5b
github.com/pilosa/pilosa/gossip.(*GossipNodeSet).GetBroadcasts(0xc420224b90, 0x3, 0x447, 0xc42024fd40, 0x3, 0x4)
	/home/ubuntu/go/src/github.com/pilosa/pilosa/gossip/gossip.go:213 +0x43
github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*Memberlist).getBroadcasts(0xc4202d0000, 0x2, 0x576, 0x4, 0xc420034e28, 0xc42027b4e0)
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/broadcast.go:88 +0xfc
github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*Memberlist).gossip(0xc4202d0000)
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/state.go:494 +0x223
github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*Memberlist).(github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.gossip)-fm()
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/state.go:94 +0x2a
github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*Memberlist).triggerFunc(0xc4202d0000, 0x5f5e100, 0xc42028c3c0, 0xc42028c300, 0xc420294260)
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/state.go:118 +0x149
created by github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist.(*Memberlist).schedule
	/home/ubuntu/go/src/github.com/pilosa/pilosa/vendor/github.com/hashicorp/memberlist/state.go:94 +0x1ed
```

Looks like GossipNodeSet.Open sets things up in the wrong order. I'll submit a PR with what I think needs to happen.

### Expected behavior
No crash.

### Actual behavior
A node crashes.

### Steps to reproduce the behavior
Not sure it's easily reproducible.

### Information about your environment (OS/architecture, CPU, RAM, cluster/solo, configuration, etc.)
Three node ubuntu cluster created by qaz in infrastructure repo.
Pilosa v0.6.0-19-g2fea07f

--- a/gossip/gossip.go
+++ b/gossip/gossip.go
@@ -69,18 +69,18 @@ func (g *GossipNodeSet) Open() error {
 		return err
 	}
 	g.memberlist = ml
+	g.broadcasts = &memberlist.TransmitLimitedQueue{
+		NumNodes: func() int {
+			return ml.NumMembers()
+		},
+		RetransmitMult: 3,
+	}
 
 	// attach to gossip seed node
 	nodes := []*pilosa.Node{&pilosa.Node{Host: g.config.gossipSeed}} //TODO: support a list of seeds
 	err = g.joinWithRetry(pilosa.Nodes(nodes).Hosts())
 	if err != nil {
 		return err
-	}
-	g.broadcasts = &memberlist.TransmitLimitedQueue{
-		NumNodes: func() int {
-			return ml.NumMembers()
-		},
-		RetransmitMult: 3,
 	}
 	return nil
 }

